crates/ruff_linter/src/rules/ruff/rules/parenthesize_logical_operators.rs

```diff
@@ -75,21 +86,47 @@ pub(crate) fn parenthesize_chained_logical_operators(
                     ..
                 },
             ) => {
+                let locator = checker.locator();
+                let source_range = bool_op.range();
                 if parenthesized_range(
                     bool_op.into(),
                     expr.into(),
                     checker.indexer().comment_ranges(),
-                    checker.locator().contents(),
+                    locator.contents(),
                 )
                 .is_none()
                 {
-                    checker.diagnostics.push(Diagnostic::new(
-                        ParenthesizeChainedOperators,
-                        bool_op.range(),
-                    ));
+                    let mut diagnostic =
+                        Diagnostic::new(ParenthesizeChainedOperators, source_range);
+                    if is_single_line_expr_with_narrow_width(
+                        locator,
+                        source_range,
+                        checker.settings,
+                    ) {
+                        let new_source = format!("({})", locator.slice(source_range));
+                        diagnostic.set_fix(Fix::applicable_edit(
```

Comment:
Nit: you can use `Fix::safe_edit` here rather than `Fix::applicable_edit` with `Applicability::Safe`.